cql3/statements/create_table_statement.cc

```diff
@@ -452,6 +453,20 @@ std::optional<sstring> check_restricted_table_properties(
             break;
         }
     }
+    if (strategy && *strategy == sstables::compaction_strategy_type::time_window) {
+        std::map<sstring, sstring> options = cfprops.get_compaction_type_options();
+        sstables::time_window_compaction_strategy_options _twcs_options(options);
+        auto ttl = cfprops.get_default_time_to_live();
+        int max_windows = qp.db().get_config().twcs_max_window_count();
+
+        if (ttl > 0) {
+            auto window_count = std::chrono::seconds(ttl) / _twcs_options.get_sstable_window_size();
+            if (max_windows > 0 && window_count > max_windows) {
+                throw exceptions::syntax_exception(fmt::format("The number of windows ({}) specified is larger than the allowed "
+                                                   "number of windows specified by the twcs_max_window_count ({}) parameter.", window_count, max_windows));
```

Comment:
I think the error message could be, and should be, more explicit on what the user set up wrong. Since this test is only about the default_ttl case, I think the error message should tell you more directly what was tested. Something like: **"The setting of default_time_to_live={} and compaction window={} can lead to {} windows, which is larger than the allowed number of windows specified by the twcs_max_window_count ({}) configuration)."**